app/models/manageiq/providers/ibm_power_hmc/infra_manager/vm.rb

```diff
@@ -1,49 +1,36 @@
 class ManageIQ::Providers::IbmPowerHmc::InfraManager::Vm < ManageIQ::Providers::InfraManager::Vm
   def provider_object(connection = nil)
-    # connection ||= ext_management_system.connect
-    # connection.find_vm(ems_ref)
+    connection ||= ext_management_system.connect
+    if vios?
+      connection.vios(ems_ref)
+    else
+      connection.lpar(ems_ref)
+    end
   end
 
   def raw_start
     $ibm_power_hmc_log.info("raw_start ems_ref=#{ems_ref}")
-    ext_management_system.with_provider_connection do |connection|
-      # Damien: check VIOS or LPAR from description?
-      connection.poweron_lpar(ems_ref)
-    end
+    power
   end
 
   def raw_stop
     $ibm_power_hmc_log.info("raw_stop ems_ref=#{ems_ref}")
-    ext_management_system.with_provider_connection do |connection|
-      # Damien: check VIOS or LPAR from description?
-      connection.poweroff_lpar(ems_ref, {"operation" => "shutdown"})
-    end
+    power({"operation" => "shutdown"})
```

Comment:
I don't know why the bot is not flagging this, but the curly braces are redundant

```suggestion
    power("operation" => "shutdown")
```

Same with the other calls below.